src/main/java/seedu/address/storage/JsonAdaptedProfessor.java

```diff
@@ -94,7 +96,18 @@ public Person toModelType() throws IllegalValueException {
         final ModuleCode modelModuleCode = new ModuleCode(getModuleCode());
 
         final Set<Tag> modelTags = new HashSet<>(personTags);
-        return new Professor(modelName, modelModuleCode, modelPhone, modelEmail, modelGender, modelTags);
+
+        if (getLocation() == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Location.class.getSimpleName()));
+        }
```

Comment:
I'm not too familiar with JSON stuff, but maybe this is not really necessary since Location is an optional field? I'm sorry if I'm mistaken.

Reply:
Yes, I agree with @normkoh location being null should not throw an error as location should be a optional field.

Reply:
I agree location could be an optional field, let's discuss about the future of this location field at our next meeting? For this iteration, I made location a compulsory field just like the other fields (name, email etc.).

Reply:
I am ok with it as long as we switch location to an optional field in the future. 😄